### PR TITLE
Separate license from header creation

### PR DESCRIPTION
**Pull request by britter**
Fix for #77 and preparation for #73. After we have separated license texts from creation of licenses, we can implement auto detection.

Note that we will probably change our settings. Currently we have:

```scala
val headers: SettingKey[Map[String, (Regex, String)]]
```

Where the key is the file extension an the tuple is the result of applying the old license trait. In the future (after I have implemented #73) we will have:

```scala
val headerLicense: SettingKey[License] // auto detected from project settings if possible
val headerMappings: SettingKey[Map[String, CommentStyle]]
```

The `headerMappings` will map file extensions to the CommentStyle to apply. I'm planning to add sensible defaults for this so that for a plain old scala project users won't need to configure anything at all.

### CODE REVIEW

File: src/main/scala/de/heikoseeberger/sbtheader/License.scala [hunk 1]
```diff
@@ -16,25 +16,13 @@
...
-  }
-
-  def createLicenseText(yyyy: String, copyrightOwner: String): String
+  def text: String
 }
 
 trait Licenses {
-  object Apache2_0 extends License {
-    override def createLicenseText(yyyy: String, copyrightOwner: String) =
+  case class Apache2_0(yyyy: String, copyrightOwner: String) extends License {
```
Comment (hseeberger) [inline src/main/scala/de/heikoseeberger/sbtheader/License.scala]: Make this case class and all other ones `final`.

**Comment by hseeberger** [comment]
Nice!

File: src/main/scala/de/heikoseeberger/sbtheader/License.scala (hunk 1 above)
Comment (britter) [inline src/main/scala/de/heikoseeberger/sbtheader/License.scala]: Damn, got that wrong *again*! Thanks for catching it.

**Comment by britter** [comment]
Codacy often breaks, because it wants field names not to contain underscores (which is the case for the `apache2_0` field).

File: src/main/scala/de/heikoseeberger/sbtheader/License.scala (hunk 1 above)
Comment (britter) [inline src/main/scala/de/heikoseeberger/sbtheader/License.scala]: @hseeberger I just realized, that compiling the code now issues warnings:

```
[warn] /Users/bene/workspace/sbt/sbt-header/src/main/scala/de/heikoseeberger/sbtheader/License.scala:24: The outer reference in this type test cannot be checked at run time.
[warn]   final case class Apache2_0(yyyy: String, copyrightOwner: String) extends License {
```

Do you think this is a problem?

**Pull request merged**